pennylane/gradients/parameter_shift_cv.py

```diff
@@ -258,6 +259,14 @@ def var_param_shift(tape, dev_wires, argnum=None, shifts=None, gradient_recipes=
     gradient_tapes.extend(pdA2_tapes)
 
     def processing_fn(results):
+        # Apply the same squeezing as in qml.QNode to make the transform output consistent.
+        # pylint: disable=protected-access
+        if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):
+            squeezed_results = []
+            for res in results:
+                squeezed_results.append(qml.math.squeeze(res))
+            results = squeezed_results

```

Comment:
It's a shame this has to be repeated in so many places :( another reason I see this as a shorter term hotfix
```suggestion
        # HOTFIX: Apply the same squeezing as in qml.QNode to make the transform output consistent.
        # pylint: disable=protected-access
        if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):
            squeezed_results = []
            for res in results:
                squeezed_results.append(qml.math.squeeze(res))
            results = squeezed_results
```